elastic/integrations
# [system] Add missing conditional checks to render `preserve_original_event` tag when toggled on

## Proposed commit message
The PR contains changes to make sure the `preserve_original_event` tag is rendered without having to add that manually. Complete details have been captured in the related issue. Affected datastreams are `application`, `security`, and `system`. 




## Checklist


## Author's Checklist
Following are some of the sanity checks that must be passed at all times:

```
no preserve_original_event and no tags ==> no tags are rendered
no preserve_original_event but tags ==> only tags should be rendered
preserve_original_event but no tags ==> only preserve_original_event is rendered under the tags section
preserve_original_event and tags ==> render all tags including preserve_original_event under the tags section 
```

## How to test this PR locally
- Spin up the Elastic stack and point your `elastic-package` to the running Kibana instance
- Build the package locally and install the package using `elastic-package install --zip system-1.54.1.zip`
- Upgrade the integration to the latest version i.e. ` 1.54.1`, there shouldn't be any errors. If you find any errors while upgrading, please report it by adding a comment to this PR.
- Add the integration to an agent policy and notice the tags sections `application`, `security`, and `system` datastream. The preserve_original_event should be rendered when toggled on, and all others tags should render if added in the advanced options at the time of adding the integration. 


## Related issues


- Relates to #9425 

## Screenshots


![image](https://github.com/elastic/integrations/assets/38116245/7d5f8a3a-370a-48db-8312-6c0b80ea19d7) 
![image](https://github.com/elastic/integrations/assets/38116245/d534b061-d771-4228-b0e0-5ad9a0512dca)![image](https://github.com/elastic/integrations/assets/38116245/3aabc9ce-043f-45b3-af2e-d16e5d78b2ef)
